File: app/intent/src/main/java/com/example/intent/SubActivity.java
package com.example.intent;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.widget.TextView;

public class SubActivity extends AppCompatActivity {

    private TextView tv_sub;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_sub);

        tv_sub = findViewById(R.id.tv_sub);

        Intent intent = getIntent();    // 이쪽으로 날아오는 값이 있으면 여기서 받겠다
        String str = intent.getStringExtra("str");   // 쏘는 곳과 동일하게 별명 맞춰줘야함


        tv_sub.setText(str);

    }
}

File: Kotlin_Practice/IntentKt/app/src/main/java/com/example/intentkt/SubActivity.kt
package com.example.intentkt

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import kotlinx.android.synthetic.main.activity_sub.*

class SubActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_sub)

        if(intent.hasExtra("msg")){ //msg라는 객체를 가지고 있으면
            tv_getMsg.text = intent.getStringExtra("msg") // 서브 액티비티에 존재하는 텍스트뷰에다가 HelloWorld가 넘겨져 옴.
        }
    }
}